core/src/main/scala/scala/pickling/generator/Macros.scala

```diff
@@ -0,0 +1,98 @@
+package scala.pickling
+package generator
+
+// TODO - move these into Pickler/Unpickler and replace the existing macros.
+private[pickling] object PicklingMacros {
+  import scala.language.experimental.macros
+  def genPickler[T]: Pickler[T] with Generated = macro scala.pickling.generator.Compat.genPickler_impl[T]
+  def genUnpickler[T]: Unpickler[T] with Generated = macro scala.pickling.generator.Compat.genUnpickler_impl[T]
+  def genPicklerUnpickler[T]: AbstractPicklerUnpickler[T] with Generated = macro scala.pickling.generator.Compat.genPicklerUnpickler_impl[T]
+}
+private[pickling] trait PicklingMacros extends Macro with SourceGenerator with TypeAnalysis {
+  import c.universe._
+  val symbols = new IrScalaSymbols[c.universe.type, c.type](c.universe, tools)
+  // TODO - We should have more customization than "isStaticOnly"
```

Comment:
+1 on this todo.
Even if we only allow `AlgorithmOption(false, false, false)` and `AlgorithmOption(true, true, true)`, I would like to see this split up to better understand the implications code gen, lookup etc.. what else?